src/functTest/groovy/com/bmuschko/gradle/docker/DockerRemoteApiPluginFunctionalTest.groovy

```diff
@@ -74,6 +81,52 @@ class DockerRemoteApiPluginFunctionalTest extends AbstractGroovyDslFunctionalTes
         build('verify')
     }
 
+    @Requires({ TestPrecondition.DOCKER_PRIVATE_SECURE_REGISTRY_REACHABLE })
+    def "can push and pull images overriding wrong registryCredentials"() {
```

Comment:
Let's also add another test case that sets the credentials via the `docker` extension instead of the tasks themselves.